tests/test_sticker.py

```diff
@@ -804,12 +810,13 @@ async def test_bot_methods_3_png(self, bot, chat_id, sticker_set_thumb_file):
     async def test_bot_methods_3_tgs(
         self, bot, chat_id, animated_sticker_file, animated_sticker_set
     ):
-        await asyncio.sleep(1)
         animated_test = f"animated_test_by_{bot.username}"
-        assert await bot.set_sticker_set_thumb(animated_test, chat_id, animated_sticker_file)
         file_id = animated_sticker_set.stickers[-1].file_id
-        # also test with file input and mask
-        assert await bot.set_sticker_set_thumb(animated_test, chat_id, file_id)
+        tasks = asyncio.gather(
+            bot.set_sticker_set_thumb(animated_test, chat_id, animated_sticker_file),
+            bot.set_sticker_set_thumb(animated_test, chat_id, file_id),
+        )
+        assert all(await tasks)
```

Comment:
both requests make changes to the same sticker set. awaiting them simultaniously could lead to strange situations. Please don't gather them

Reply:
in this test we only we checked if the sticker thumb was set, by checking the boolean result of it. What strange situations do you think would happen?

Reply:
Req_1 start
Req_2 start
Req_1 finish
Req_2 finish

What happens in this scenario? Does TG care about the starting of the requests or does it just treat that is "req_1 came in and then req_2 came in"? The chance of something going wrong here is probably very low but so is the advantage of gathering  IMO …

Reply:
> What happens in this scenario

nothing... TG just sees whatever request came first to them, and would set that as the thumb. The test has never failed for me for that reason... I could still revert it if you'd like

Reply:
My gut feeling is not to use `gather` here. But I won't veto.